In [14]:
import pandas as pd

df = pd.read_csv(r'hasta_kayitlari.csv')

### Bölüm 1: Veriyi Tanımak

In [15]:
df.head(8)

,hasta_id,yas,cinsiyet,sehir,kan_grubu,bolum,tani,yatis_gun,bmi,sistolik_kb,sigorta,maliyet_tl
0,10630,49,Kadın,İzmir,0-,Nöroloji,İnme,2,19.2,103.0,Yok,4882.0
1,10789,29,Kadın,İstanbul,AB+,Kardiyoloji,Hipertansiyon,3,NaN,126.0,SGK,10119.0
2,10685,69,Erkek,İstanbul,A+,Dahiliye,Anemi,7,28.1,124.0,SGK,9346.0
3,10517,19,Kadın,Konya,A+,Pediatri,Bronşit,1,23.7,94.0,Özel Sigorta,3094.0
4,10530,77,Kadın,İstanbul,0+,Kardiyoloji,Hipertansiyon,2,26.4,135.0,SGK,6823.0
5,10658,75,Kadın,Ankara,A+,Pediatri,Bronşit,7,27.5,135.0,Özel Sigorta,8981.0
6,10553,83,Kadın,İzmir,0+,Pediatri,Bronşit,1,27.2,126.0,Özel Sigorta,3377.0
7,10532,66,Kadın,İstanbul,AB+,Ortopedi,Kırık,1,27.0,NaN,SGK,5569.0


In [16]:
df.tail(4)

,hasta_id,yas,cinsiyet,sehir,kan_grubu,bolum,tani,yatis_gun,bmi,sistolik_kb,sigorta,maliyet_tl
1006,10271,18,Erkek,Ankara,0+,Ortopedi,Tendon Yırtığı,2,22.1,118.0,SGK,8435.0
1007,10861,59,Kadın,Ankara,0+,Dahiliye,Anemi,7,28.9,132.0,SGK,9554.0
1008,10436,49,Kadın,İstanbul,AB+,Dahiliye,Tiroid,4,23.5,112.0,Özel Sigorta,7420.0
1009,10103,77,Erkek,Konya,A+,Kardiyoloji,Hipertansiyon,7,27.3,137.0,SGK,14731.0


In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1010 entries, 0 to 1009
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   hasta_id     1010 non-null   int64  
 1   yas          1010 non-null   int64  
 2   cinsiyet     1010 non-null   object 
 3   sehir        1010 non-null   object 
 4   kan_grubu    1010 non-null   object 
 5   bolum        1010 non-null   object 
 6   tani         1010 non-null   object 
 7   yatis_gun    1010 non-null   int64  
 8   bmi          962 non-null    float64
 9   sistolik_kb  995 non-null    float64
 10  sigorta      1010 non-null   object 
 11  maliyet_tl   988 non-null    float64
dtypes: float64(3), int64(3), object(6)
memory usage: 94.8+ KB


bmi, sistolik ve maliyet_tl verilerinde sırasıyla 48, 15 ve 22 veri eksik.

In [18]:
df.rename(columns={"maliyet_tl":"maliyet"},inplace=True)
df.rename(columns={"yatis_gun":"yatis_suresi"},inplace=True)
df.head(3)

,hasta_id,yas,cinsiyet,sehir,kan_grubu,bolum,tani,yatis_suresi,bmi,sistolik_kb,sigorta,maliyet
0,10630,49,Kadın,İzmir,0-,Nöroloji,İnme,2,19.2,103.0,Yok,4882.0
1,10789,29,Kadın,İstanbul,AB+,Kardiyoloji,Hipertansiyon,3,NaN,126.0,SGK,10119.0
2,10685,69,Erkek,İstanbul,A+,Dahiliye,Anemi,7,28.1,124.0,SGK,9346.0


### Bölüm 2: Veri Temizleme

In [19]:
print("Tekrar eden veriler silinmeden önce uzunluk: ", len(df))
df.duplicated().sum()

Tekrar eden veriler silinmeden önce uzunluk:  1010


np.int64(10)

In [20]:
df=df.drop_duplicates()
print("Tekrar eden veriler silindikten sonra uzunluk: ", len(df))

Tekrar eden veriler silindikten sonra uzunluk:  1000


Sütunlardaki eksik veri miktarlarını listeleyelim

In [21]:
print(df.isna().sum())

hasta_id         0
yas              0
cinsiyet         0
sehir            0
kan_grubu        0
bolum            0
tani             0
yatis_suresi     0
bmi             48
sistolik_kb     15
sigorta          0
maliyet         22
dtype: int64


bmi, sistolik ve maliyet_tl verilerinde sırasıyla 48, 15 ve 22 veri eksik.

In [22]:
df['bmi'] = df['bmi'].fillna(df['bmi'].median())
print(df.isna().sum())

hasta_id         0
yas              0
cinsiyet         0
sehir            0
kan_grubu        0
bolum            0
tani             0
yatis_suresi     0
bmi              0
sistolik_kb     15
sigorta          0
maliyet         22
dtype: int64


In [23]:
df['sistolik_kb'] = df['sistolik_kb'].fillna(df['sistolik_kb'].median())
print(df.isna().sum())

hasta_id         0
yas              0
cinsiyet         0
sehir            0
kan_grubu        0
bolum            0
tani             0
yatis_suresi     0
bmi              0
sistolik_kb      0
sigorta          0
maliyet         22
dtype: int64


In [28]:
df = df.dropna(subset=['maliyet'])
print(df.isna().sum())

hasta_id        0
yas             0
cinsiyet        0
sehir           0
kan_grubu       0
bolum           0
tani            0
yatis_suresi    0
bmi             0
sistolik_kb     0
sigorta         0
maliyet         0
dtype: int64


In [29]:
len(df)

978

### Bölüm 3: Filtreleme ve Yeni Sütun Oluşturma

In [30]:
df[(df.sehir=="İstanbul") & (df.yas>50)]

,hasta_id,yas,cinsiyet,sehir,kan_grubu,bolum,tani,yatis_suresi,bmi,sistolik_kb,sigorta,maliyet
2,10685,69,Erkek,İstanbul,A+,Dahiliye,Anemi,7,28.1,124.0,SGK,9346.0
4,10530,77,Kadın,İstanbul,0+,Kardiyoloji,Hipertansiyon,2,26.4,135.0,SGK,6823.0
7,10532,66,Kadın,İstanbul,AB+,Ortopedi,Kırık,1,27.0,122.0,SGK,5569.0
9,10071,78,Erkek,İstanbul,B+,Genel Cerrahi,Kolon Polip,1,28.4,136.0,SGK,5760.0
19,10102,65,Erkek,İstanbul,0+,Kardiyoloji,Aritmia,3,21.2,121.0,SGK,7377.0
...,...,...,...,...,...,...,...,...,...,...,...,...
953,10991,65,Kadın,İstanbul,A-,Nöroloji,İnme,4,24.1,134.0,Özel Sigorta,8922.0
958,10876,74,Kadın,İstanbul,B+,Nöroloji,İnme,1,32.3,146.0,SGK,4346.0
982,10386,66,Kadın,İstanbul,A+,Nöroloji,Migren,5,27.1,130.0,SGK,10878.0
989,10662,51,Kadın,İstanbul,AB-,Kardiyoloji,Hipertansiyon,2,33.4,127.0,Yok,7272.0


İstanbul'da yaşayan ve yaşı 50'den büyük 124 kişi vardır

In [33]:
sigortasiz_yatis = df[(df.sigorta=="Yok") & (df.yatis_suresi>7)]
sigortasiz_yatis['maliyet'].mean()

np.float64(21769.53846153846)

Sigortası olmayan ve 7 günden fazla yatış yapan hastaların ortalama maliyeti 21769.53

In [40]:
df['gunluk_maliyet'] = df['maliyet'] / df['yatis_suresi']
df['yas_grubu'] = [
    'genç' if 18 < i <= 35 
    else 'orta yaşlı' if 35 < i < 60 
    else 'yaşlı'
    for i in df['yas']
]
df.head(3)

,hasta_id,yas,cinsiyet,sehir,kan_grubu,bolum,tani,yatis_suresi,bmi,sistolik_kb,sigorta,maliyet,gunluk_maliyet,yas_grubu
0,10630,49,Kadın,İzmir,0-,Nöroloji,İnme,2,19.2,103.0,Yok,4882.0,2441.000000,orta yaşlı
1,10789,29,Kadın,İstanbul,AB+,Kardiyoloji,Hipertansiyon,3,24.7,126.0,SGK,10119.0,3373.000000,genç
2,10685,69,Erkek,İstanbul,A+,Dahiliye,Anemi,7,28.1,124.0,SGK,9346.0,1335.142857,yaşlı


In [43]:
df[df['bolum'] == 'Kardiyoloji'].sort_values('maliyet', ascending=False).head(10)

,hasta_id,yas,cinsiyet,sehir,kan_grubu,bolum,tani,yatis_suresi,bmi,sistolik_kb,sigorta,maliyet,gunluk_maliyet,yas_grubu
534,10807,63,Kadın,İzmir,B-,Kardiyoloji,Aritmia,21,25.5,143.0,SGK,27636.0,1316.000000,yaşlı
314,10588,51,Kadın,Antalya,0+,Kardiyoloji,Kalp Yetmezliği,21,19.8,121.0,Yok,27571.0,1312.904762,orta yaşlı
635,10516,83,Erkek,İstanbul,B+,Kardiyoloji,Aritmia,14,34.0,137.0,SGK,22667.0,1619.071429,yaşlı
468,10105,19,Kadın,Gaziantep,B+,Kardiyoloji,Kalp Yetmezliği,14,18.7,116.0,SGK,22553.0,1610.928571,genç
492,10501,33,Erkek,Konya,B+,Kardiyoloji,Kalp Yetmezliği,14,23.5,109.0,Yok,21807.0,1557.642857,genç
750,10922,78,Erkek,İstanbul,AB+,Kardiyoloji,Hipertansiyon,10,31.2,132.0,SGK,18355.0,1835.500000,yaşlı
727,10969,42,Erkek,İzmir,A-,Kardiyoloji,Hipertansiyon,10,23.7,116.0,SGK,18298.0,1829.800000,orta yaşlı
904,10727,47,Erkek,Antalya,B+,Kardiyoloji,Aritmia,10,28.4,120.0,Özel Sigorta,18291.0,1829.100000,orta yaşlı
235,10582,60,Erkek,İstanbul,0+,Kardiyoloji,Hipertansiyon,10,27.0,128.0,SGK,18011.0,1801.100000,yaşlı
502,10891,72,Kadın,İstanbul,A+,Kardiyoloji,Koroner Arter,10,28.2,128.0,Özel Sigorta,17993.0,1799.300000,yaşlı


Kardiyoloji bölümünde en çok maliyete sahip hastalarından ilk 10 kişiyi seçtiğimizde büyük çoğunluğu sigortalıdır (8 sigortalı, 2 sigortasız). Sigortalıların çoğu da SGK'lıdır.

### Bölüm 4: Özet İstatistikleri 

In [45]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
hasta_id,978.0,10501.006135,289.072997,10001.00000,10251.25,10503.5,10749.7500,11000.0
yas,978.0,49.568507,19.155772,18.00000,33.00,49.0,66.0000,84.0
yatis_suresi,978.0,3.860941,3.628485,1.00000,2.00,3.0,5.0000,21.0
bmi,978.0,24.762065,3.369455,16.00000,22.50,24.7,26.9000,34.0
sistolik_kb,978.0,122.535787,12.266457,90.00000,114.00,122.0,131.0000,160.0
maliyet,978.0,8746.993865,5279.781504,1868.00000,5153.75,7338.5,10677.2500,37455.0
gunluk_maliyet,978.0,2914.467326,1218.735877,838.47619,1998.05,2666.7,3623.5625,6865.0


In [ ]:
df['sehir'].value_counts()

sehir
İstanbul     282
Ankara       164
İzmir        151
Bursa         93
Antalya       92
Adana         76
Konya         70
Gaziantep     50
Name: count, dtype: int64

In [55]:
df['sehir'].value_counts(normalize=True)

sehir
İstanbul     0.288344
Ankara       0.167689
İzmir        0.154397
Bursa        0.095092
Antalya      0.094070
Adana        0.077710
Konya        0.071575
Gaziantep    0.051125
Name: proportion, dtype: float64

Hasta yoğunluğu olarak en kalabalık 3 şehir sırasıyla İstanbul, Ankara ve İzmir'dir. 
Bu şehirlerin diğer şehirlere kıyasla yoğunluk oranı sırasıyla %28.9, %16.8 ve %15.4 dür.

In [56]:
df['bolum'].value_counts()

bolum
Kardiyoloji      176
Dahiliye         171
Pediatri         170
Nöroloji         160
Ortopedi         151
Genel Cerrahi    150
Name: count, dtype: int64

In [57]:
df['bolum'].value_counts(normalize=True)

bolum
Kardiyoloji      0.179959
Dahiliye         0.174847
Pediatri         0.173824
Nöroloji         0.163599
Ortopedi         0.154397
Genel Cerrahi    0.153374
Name: proportion, dtype: float64

Bölümlerin hasta yoğunlukları birbirlerine yakın olmakla birlikte en yoğun iki bölüm sırasıyla Kardiyoloji ve Dahiliye bölümleridir. Bu bölümlerin hasta sayısının toplama oranı sırasıyla %18.0 ve %17.5'dir.

In [61]:
df.groupby("bolum")[["yas", "yatis_suresi", "maliyet"]].agg(["mean", "max", "count", "sum"]).round(2).T

bolum                 Dahiliye  Genel Cerrahi  Kardiyoloji    Nöroloji  \
yas          mean        49.44          50.19        50.22       49.53   
             max         84.00          84.00        84.00       84.00   
             count      171.00         150.00       176.00      160.00   
             sum       8454.00        7528.00      8839.00     7925.00   
yatis_suresi mean         3.77           4.03         3.88        3.58   
             max         21.00          21.00        21.00       21.00   
             count      171.00         150.00       176.00      160.00   
             sum        645.00         604.00       683.00      572.00   
maliyet      mean      6531.11       12684.94      9383.80     7587.96   
             max      20520.00       37455.00     27636.00    23507.00   
             count      171.00         150.00       176.00      160.00   
             sum    1116820.00     1902741.00   1651548.00  1214074.00   

bolum                 Ortopedi    Pediatri  
yas          mean        49.03       48.99  
             max         84.00       84.00  
             count      151.00      170.00  
             sum       7404.00     8328.00  
yatis_suresi mean         3.97        3.96  
             max         21.00       21.00  
             count      151.00      170.00  
             sum        599.00      673.00  
maliyet      mean     11014.30     5918.93  
             max      32577.00    17974.00  
             count      151.00      170.00  
             sum    1663159.00  1006218.00

Her bölüm için ortalama ve maksimum maliyet yatış süresi ve yaş bulunmuştur.

In [63]:
df.groupby("yas_grubu")[["bmi"]].mean().round(2)

,bmi
yas_grubu,
genç,22.96
orta yaşlı,24.80
yaşlı,26.09


Yaş ile BMI arasında doğrusal korelasyon gözlenmiştir. Yaş arttıkça ortalama BMI artmaktadır

In [65]:
df.groupby("sigorta")[["yatis_suresi", "maliyet"]].mean().round(2).T

sigorta,SGK,Yok,Özel Sigorta
yatis_suresi,3.86,4.02,3.78
maliyet,8706.79,9095.41,8627.39


Sigortasıs hastaların maliyeti sigortalılara kıyasla daha fazladır.

In [73]:
df[df["sigorta"] == "Yok"].groupby("bolum").size().sort_values(ascending=False)

bolum
Kardiyoloji      34
Ortopedi         28
Dahiliye         27
Pediatri         26
Nöroloji         23
Genel Cerrahi    21
dtype: int64

En çok sigortasız hasta Kardiyoloji bölümünde bulunmaktadır.